### DISCUSSION TITLE
RenderOptions in Render() ignored when visual parent(s) have RenderOptions set in XAML

### DISCUSSION THREAD

**Discussion by AngryCarrot789** [AvaloniaUI/Avalonia#17773] [Q&A]
I'm not sure why this behaviour is happening but this is my ControlTemplate: 
```
<ControlTemplate>
    <Border x:Name="PART_ClipBorderRoot" BorderThickness="1,0"
            BorderBrush="{DynamicResource ABrush.Tone1.Border.Static}"
            UseLayoutRounding="False" RenderOptions.EdgeMode="Aliased">
        <Panel x:Name="PART_Panel" Margin="0,20,0,0">
            <ContentPresenter x:Name="PART_ContentPresenter"
                              HorizontalAlignment="Stretch" VerticalAlignment="Stretch"
                              Content="{TemplateBinding Content}"
                              ContentTemplate="{TemplateBinding ContentTemplate}"
                              CornerRadius="{TemplateBinding CornerRadius}"
                              RecognizesAccessKey="True"
                              TextElement.Foreground="{TemplateBinding Foreground}"/>
            <a:AutomationSequenceEditorControl x:Name="PART_AutomationEditor" Margin="0,5"/>
        </Panel>
    </Border>
</ControlTemplate>
```

And in my `AutomationSequenceEditorControl `'s Render method I'm pushing render options with anti-aliasing onto the stack, but the pixels are not anti-aliased. Anyone know why this is happening?

This is what I currently get: 
![image](https://github.com/user-attachments/assets/280b8b71-fe65-4ff1-9404-94b13aafee1e)

But if I remove the `RenderOptions.EdgeMode` from the Border, I get normal anti-aliasing except the control I've made now doesn't render quite right, I get weird sub pixel issues around the border: 
![image](https://github.com/user-attachments/assets/581b08aa-4be3-42e9-bb1f-6b1d89c386fe)

**Comment by stevemonaco** [comment]
Can you show code to demonstrate how you're pushing the new `RenderOptions`?

I can't see any issue on the border.

**Comment by AngryCarrot789** [comment]
The border is only rendered weird when I select the control, since the BorderThickness becomes 2. The code is literally just something like `using var unused = ctx.PushRenderOpton(new RenderOptions() { EdgeMode = EdgeMode.Antialias })`

**Reply by timunie** [reply]
New renderootions, that's it probably. Better to use the .WithXxx method

**Reply by AngryCarrot789** [reply]
I get the exact same behaviour if I set the EdgeMode for AutomationSequenceEditorControl in the XAML too which I assume will use WithEdgeMode in the background
